### 7.2. Calibration Results

This is organized in the following way

- [MATLAB calibration routine](#cal)

- [Zeta normalization analysis](#zeta)

In [6]:
julia_path = '''C:\\Users\\mwaugh\\AppData\\Local\\Programs\\Julia\\Julia-1.4.0\\bin\\julia.exe'''

matlab_path = '''matlab'''
# This assumes that Matlab is in your path. If not then direct it to
# program files / matlab / version/ bin/ matlab.exe
# needs to be in triple quotes

---
### <a name="cal"></a> Calibration Routine

The code below access the commandline/terminal and then directly executes the MATLAB code. It assumes that (i) you have a working installation of MATLAB and (ii) that it is in your path. Also verify that the directory change implemented below is correct.

The resulting parameters are outputed to [``/parameters/calibration_params.csv``](/parameters/calibration_params.csv)

The underlying MATLAB code is completly described (with links to the appropriate ``.m`` files) in the [readme file in the calibraiton folder](/src/calibration/README.md).

In [7]:
matlab_cmd = '''"cd('src\calibration');calibrate_wrap;"'''
# then the way this works is you have to give the matlab command to change the working directory
# this is the cd part; then call the function which in this case is calibrate_wrap
# needs to be in triple quotes 

In [8]:
!{matlab_path} -batch {matlab_cmd}

# Then the ! command passes the stuff to the command prompt
# or terminal. the {} remove the triple quotes
# -batch runs this stuff in the background, i.e. no matlab
# terminal is opend, it will autormatically exit and close matlab
# then the next {} is function call to matlab

   20-Apr-2020

Calibration Targets...
Real Rate and Productivity Growth
    0.0283    0.0079

BEJK Exporter Moments: Fraction of Exporters, Relative Size
    0.0330    4.8000

Home Trade Share
    0.8937

Entry Moment
    0.0200

Firm Moments (LHS Table 3)
    0.5010    0.2700    0.1360    0.0940
    0.1610    0.2850    0.2900    0.2600

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
 
Optimization terminated:
 the current x satisfies the termination criteria using OPTIONS.TolX of 1.000000e-04 
 and F(X) satisfies the convergence criteria using OPTIONS.TolFun of 1.000000e-04 

Elapsed time is 39.050084 seconds.
    0.0037

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
Calibration Results
Parameter Values
d, theta, kappa, 1/chi, mu, upsilon, sigma, delta, rho
    3.0225    4.9890    0.1042    7.8835   -0.0311    0.0483    3.1669    0.0200    0.0203

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
Mom

Again, output is at [``/parameters/calibration_params.csv``](/parameters/calibration_params.csv) which contains the parameter values in the with headers for each column describing the associated parameter. Order should be the same as in [``/src/full/params.jl``](/src/full/params.jl).

#### Brief Commentary (same as in paper)

In terms of fit, the aggregate moments (real rate and productivity growth), the exporter facts, and the employment share of entrants match
the data exactly. The model’s fit for the eight firm dynamic moments are the final outputs  While not perfect, the model
replicates the dynamics of firms well. The correlation between the model and the data is extremely high at 0.98.

---
### <a name="zeta"></a> Zeta Normalization Analysis

In Section 7.2, we discuss our normalization of zeta and how our results do not depend on the particular normalization. The discussion is below...

As stated earlier, we set ζ = 1 as a normalization. Several comments are in order regarding this normalization. First, calibrating the model with a different normalization results in identical parameter values for all parameters except κ. In the theoretical analysis, equations (33) and (35) show that ζ only appears in the form ζ/κ. Thus, a different choice of ζ results in a different κ to match the g and λii moments.
Different normalizations do lead to a different amount of variety as equation (48) shows. However, as equation (53) shows, any combination of ζ and κ results in the same welfare gains from trade as long as λii and all other parameter values are equal. It is in this sense that ζ = 1 is a normalization. Numerical analysis of the quantitative model confirms these observations both for steady-state comparative statics
and for the analysis of transition dynamics

#### Calibration under alternative  ζ

In [9]:
matlab_cmd = '''"cd('src\calibration');calibrate_wrap_zeta;"'''
# then the way this works is you have to give the matlab command to change the working directory
# this is the cd part; then call the function which in this case is calibrate_wrap_zeta
# needs to be in triple quotes 

In [10]:
!{matlab_path} -batch {matlab_cmd}

Calbirated values computed on date
   20-Apr-2020

Calibration with different zetas
Zeta Value
    0.7500

 
Optimization terminated:
 the current x satisfies the termination criteria using OPTIONS.TolX of 1.000000e-04 
 and F(X) satisfies the convergence criteria using OPTIONS.TolFun of 1.000000e-04 

Elapsed time is 37.550956 seconds.
Parameter Values
d, theta, kappa, 1/chi, mu, upsilon, sigma, delta, rho, zeta
    3.0225    4.9890    0.0781    7.8835   -0.0311    0.0483    3.1669    0.0200    0.0203    0.7500

zeta/kappa
    9.5973

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
Moments: Model Alt Zetas
Calibration Targets...
Real Rate and Productivity Growth
    0.0283    0.0079

BEJK Exporter Moments: Fraction of Exporters, Relative Size
    0.0330    4.7999

Home Trade Share
    0.8937

Entry Moment
    0.0200

Firm Moments
    0.2587    0.2883    0.0861    0.3148
    0.2823    0.1706    0.1418    0.4573

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

Note the calibrated parameter values are all in [``/parameters/calibration_zeta_X.XX.csv``](/parameters) and each ``.csv`` has the final endings of XX which coresponds with the zeta value assumed. Headers describe the parameters, order should be the same as in [``/src/full/params.jl``](/src/full/params.jl).

#### Brief Commentary

Per our claim in the paper: The calibration results lead to (i) the same exact fit (ii) all parameter values are the same except for kappa and (iii) the ratio of zeta/kappa is always the same (9.5973)

The next set of code confirms that this does not matter for counterfactualls or welfare.

#### Alternative zetas: counterfactuals and welfare.

In [11]:
!jupyter nbconvert --to script ZetaNormalization.ipynb

julia_command = '''ZetaNormalization.jl'''

!{julia_path} {julia_command}

[NbConvertApp] Converting notebook ZetaNormalization.ipynb to script
[NbConvertApp] Writing 3699 bytes to ZetaNormalization.jl


-------------------------------------------------------------------------
Printing out pramater, Zeta = 1
Baseline (Zeta = 1) vs. 10% Lower Trade Costs


Change in Growth, Steady State to Steady State
Growth in Baseline: 0.79
Growth in New SS: 1.03


Zeta = 1 Imports/GDP, Steady State to Steady State
Imports/GDP in Baseline: 10.6
Imports/GDP in New SS: 14.4


Zeta = 1 Consumption Equivalent Welfare Gain, Steady State to Steady State
Zeta = 1, CE gains: 11.2

-------------------------------------------------------------------------
Printing out pramater, Zeta = 0.75
Zeta = 0.75 vs. 10% Lower Trade Costs


Change in Growth, Steady State to Steady State
Growth with Zeta = 0.75:  0.79
Growth in New SS:  1.03


Zeta = 0.75, Imports/GDP, Steady State to Steady State
Imports/GDP,  Zeta = 0.75: 10.6
Imports/GDP in New SS: 14.4


Zeta = 0.75 Consumption Equivalent Welfare Gain, Steady State to Steady State
Zeta = 0.75 CE gains: 11.2

-------------------------------------------------------------

As the results illustrate they show that the both (i) initial allocations for growth and trade are the same (ii) the change in growth and trade are the same and (iii) the welfare gain is the same across all the different example normalizations. 